applications/DEM_application/custom_utilities/analytic_tools/analytic_face_watcher.cpp

```diff
@@ -0,0 +1,155 @@
+            std::vector<int> colliding_ids = face.GetIdsOfCrossers();
+            std::vector<double> colliding_normal_vel = face.GetCollidingNormalRelativeVelocity();
+            std::vector<double> colliding_tangential_vel = face.GetCollidingTangentialRelativeVelocity();
+
+            for (int i = 0; i < n_collisions; ++i){
+                time_step_database.PushBackCrossings(id, colliding_ids[i], colliding_normal_vel[i], colliding_tangential_vel[i]);
+                face_database.PushBackCrossings(current_time, colliding_ids[i], colliding_normal_vel[i], colliding_tangential_vel[i]);
+            }
+        }
+    }
+
+    mVectorOfTimeStepDatabases.push_back(time_step_database);
+
+    if (time_step_database.GetNumberOfCrossings()){
+
+    }
+}
+
+void AnalyticFaceWatcher::ClearList(boost::python::list& my_list)
+{
+    while (len(my_list)){
+        my_list.pop(); // only way I found to remove all entries
+    }
+}
+
+void AnalyticFaceWatcher::GetFaceData(int id,
+                                      boost::python::list times,
+                                      boost::python::list neighbour_ids,
+                                      boost::python::list normal_relative_vel,
+                                      boost::python::list tangential_relative_vel)
+{
+    mMapOfFaceHistoryDatabases[id].FillUpPythonLists(times, neighbour_ids, normal_relative_vel, tangential_relative_vel);
+}
+
+void AnalyticFaceWatcher::GetAllFacesData(ModelPart& analytic_model_part,
+                                          boost::python::list times,
+                                          boost::python::list neighbour_ids,
+                                          boost::python::list normal_relative_vel,
+                                          boost::python::list tangential_relative_vel)
+{
+    ClearList(times);
+    ClearList(neighbour_ids);
+    ClearList(normal_relative_vel);
+    ClearList(tangential_relative_vel);
+
+    for (ConditionsIteratorType i_cond = analytic_model_part.ConditionsBegin(); i_cond != analytic_model_part.ConditionsEnd(); ++i_cond){
+        boost::python::list times_i;
+        boost::python::list neighbour_ids_i;
+        boost::python::list normal_relative_vel_i;
+        boost::python::list tangential_relative_vel_i;
+        const int id = int(i_cond->Id());
+        GetFaceData(id, times_i, neighbour_ids_i, normal_relative_vel_i, tangential_relative_vel_i);
+        times.append(times_i);
+        neighbour_ids.append(neighbour_ids_i);
+        normal_relative_vel.append(normal_relative_vel_i);
+        tangential_relative_vel.append(tangential_relative_vel_i);
+    }
+
+}
+
+void AnalyticFaceWatcher::GetTimeStepsData(boost::python::list ids,
+                                           boost::python::list neighbour_ids,
+                                           boost::python::list normal_relative_vel,
+                                           boost::python::list tangential_relative_vel)
```

Comment:
Do you think adding & would make a difference? I guess these lists are working already, so they are being modified in the function, but intuitively I would have received these arguments as a reference...